source/brailleTables.py

```diff
@@ -82,325 +97,470 @@ def listTables():
+	addTable("sl-si-comp8.ctb", _("Slovenian 8 dot computer braille"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("sl-si-g1.utb", _("Slovenian grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("sr-g1.ctb", _("Serbian grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("ta-ta-g1.ctb", _("Tamil grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("te-in-g1.utb", _("Telugu grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("tr.ctb", _("Turkish grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("uk.utb", _("Ukrainian"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("unicode-braille.utb", _("Unicode braille"), output=False)
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("vi-g1.ctb", _("Vietnamese grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zhcn-g1.ctb", _("Chinese (China, Mandarin) grade 1"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zhcn-g2.ctb", _("Chinese (China, Mandarin) grade 2"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zh-hk.ctb", _("Chinese (Hong Kong, Cantonese)"))
+	# Translators: The name of a braille table displayed in the
+	# braille settings dialog.
+	addTable("zh-tw.ctb", _("Chinese (Taiwan, Mandarin)"))
+
+	# Add custom tables.
+	import os
+	from logHandler import log
+	customDirs = _getCustomTablesDirs()
+	tablesDirs[:0] = customDirs
+	# Reversing the order ensures that user defined configuration takes
+	# precedence over addons.
+	log.warning(f"customDirs={customDirs!r}")
+	for dir_ in reversed(customDirs):
+		# Redundant check, in case an add-on added an external directory to the list.
+		if not os.path.isdir(dir_):
+			continue
+		for fileName in os.listdir(dir_):
+			path = os.path.join(dir_, fileName)
+			if os.path.isfile(path) and fileName.endswith(".ini"):
+				try:
+					with open(path, "rb") as file_:
+						manifest = CustomTablesManifest(file_)
+						_loadCustomTablesConfig(manifest.dict())
+				except:  # noqa: E722
+					log.exception(
+						"Error while applying custom braille tables config: {}"
+						"".format(path)
+					)
+
+
+def terminate():
+	global tablesDirs, _tables
+	tablesDirs = [TABLES_DIR]
+	_tables = {}
+
+
+def _getCustomTablesDirs():
+	"""Retrieve the custom braille tables directories.
+	
+	These are "brailleTables" sub-directories searched after in order in:
+	 - The scratchpad directory, if enabled.
+	 - The user configuration directory.

```

Comment:
This is no longer true.

Reply:
You are right, this one has been removed as of your pre-PR comments of the Python 2 version.